sklearn/utils/_testing.py

```diff
@@ -655,14 +655,22 @@ def check_docstring_parameters(func, doc=None, ignore=None):
 
     # Analyze function's docstring
     if doc is None:
-        with warnings.catch_warnings(record=True) as w:
+        records = []
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter("error", UserWarning)
             try:
                 doc = docscrape.FunctionDoc(func)
+            except UserWarning as exp:
+                if "potentially wrong underline length" in str(exp):
+                    message = str(exp).split("\n")[:4]
+                    incorrect += [f"In function: {func_name}"] + message
+                    return incorrect
+                records.append(str(exp))
```

Comment:
This is to catch the new UserWarning when the underline length for a section is not consistent.

Reply:
I guess this can be added as an inline comment:
```suggestion
            except UserWarning as exp:
                if "potentially wrong underline length" in str(exp):
                    # Catch warning raised as of numpydoc 1.2 when
                    # the underline length for a section of a docstring
                    # is not consistent.
                    message = str(exp).split("\n")[:4]
                    incorrect += [f"In function: {func_name}"] + message
                    return incorrect
                records.append(str(exp))
```